src/Infrastructure/Repositiory/PaymentRepository.cs

```diff
@@ -0,0 +1,51 @@
+using Application.Interfaces.Repositories;
+using Domain;
+
+namespace Infrastructure.Repositories;
+
+public class PaymentRepository : IPaymentRepository
+{
+    private readonly List<Payment> _paymentCollection;
+    public PaymentRepository()
+    {
+        _paymentCollection = new();
+    }
+
+    public void Add(Payment model)
+    {
+        long currentID = _paymentCollection.Count;
+        currentID++;
+        Payment newPayment = model with { ID = currentID };
+        _paymentCollection.Add(newPayment);
+    }
+
+    public void Delete(long id)
+    {
+        Payment paymentToDelete = new Payment { ID = id, Status = 'D' };
+        if (paymentToDelete is not null)
+            Update(paymentToDelete);
+    }
+
+    public List<Payment> GetAll()
+    {
+        return _paymentCollection.Where(p => p.Status == 'A').ToList();
+    }
+
+    public Payment? GetByID(long id)
+    {
+        return _paymentCollection?.FirstOrDefault(p => p.Status == 'A' && p.ID == id);
+    }
+
+    public void Update(Payment model)
```

Comment:
What would happen if I just want to update ViaOfPayment?

Also I would suggest tl rename to `method`. Like payment method

Reply:
1. Right now, the way is designed requires to pass a full copy of the original entity with the modification.

ex:
```csharp
Product myEntoty = getProductById(1); //Get the entity to update.

var productToUpdate = myEntoty with {Name="new entity name"}; //Create a copy of the original but changing the value of the property `Name`

repo.Update(productToUpdate);
```
2. I'll rename it.

Reply:
Then if that will always happen then why you need to duplicate it in the first place?

Why does the developer needs to do it? Is it not easier to simplify that to the dev?

Reply:
In fact, the data will be not duplicated. Record by default are immutable, so I can't just update one field, I need to replace the full record. That's why I pass a copy with the modification.

This scenario is different using a persistence system, like a database. Since I'm using a list as a data store (for now) it requires passing the whole entity.